From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 18 Sep 2017 20:22:57 +0200
Subject: [PATCH] {For LLVMLinux} drivers: make container_of compatible with
 clang

ERROR: "__compiletime_assert_1462" [drivers/media/platform/qcom/camss-8x16/qcom-camss.ko] undefined!
---
 drivers/media/platform/qcom/camss-8x16/camss-ispif.c | 4 ++--
 drivers/media/platform/qcom/camss-8x16/camss-vfe.c   | 6 +++---
 drivers/media/platform/qcom/camss-8x16/camss.h       | 4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/media/platform/qcom/camss-8x16/camss-ispif.c b/drivers/media/platform/qcom/camss-8x16/camss-ispif.c
index 24da529397b5..b2c971b66d04 100644
--- a/drivers/media/platform/qcom/camss-8x16/camss-ispif.c
+++ b/drivers/media/platform/qcom/camss-8x16/camss-ispif.c
@@ -32,10 +32,10 @@
 #define MSM_ISPIF_NAME "msm_ispif"
 
 #define ispif_line_array(ptr_line)	\
-	((const struct ispif_line (*)[]) &(ptr_line[-(ptr_line->id)]))
+	((const struct ispif_line *) &(ptr_line[-(ptr_line->id)]))
 
 #define to_ispif(ptr_line)	\
-	container_of(ispif_line_array(ptr_line), struct ispif_device, ptr_line)
+	container_of(ispif_line_array(ptr_line), struct ispif_device, ptr_line[0])
 
 #define ISPIF_RST_CMD_0			0x008
 #define ISPIF_RST_CMD_0_STROBED_RST_EN		(1 << 0)
diff --git a/drivers/media/platform/qcom/camss-8x16/camss-vfe.c b/drivers/media/platform/qcom/camss-8x16/camss-vfe.c
index 55232a912950..c1fc3292393b 100644
--- a/drivers/media/platform/qcom/camss-8x16/camss-vfe.c
+++ b/drivers/media/platform/qcom/camss-8x16/camss-vfe.c
@@ -35,10 +35,10 @@
 #define MSM_VFE_NAME "msm_vfe"
 
 #define vfe_line_array(ptr_line)	\
-	((const struct vfe_line (*)[]) &(ptr_line[-(ptr_line->id)]))
+	((const struct vfe_line *) &(ptr_line[-(ptr_line->id)]))
 
-#define to_vfe(ptr_line)	\
-	container_of(vfe_line_array(ptr_line), struct vfe_device, ptr_line)
+#define to_vfe(ptr_line)	({ \
+	container_of(vfe_line_array(ptr_line), struct vfe_device, ptr_line[0]); })
 
 #define VFE_0_HW_VERSION		0x000
 
diff --git a/drivers/media/platform/qcom/camss-8x16/camss.h b/drivers/media/platform/qcom/camss-8x16/camss.h
index 4ad223443e4b..db285b8df52d 100644
--- a/drivers/media/platform/qcom/camss-8x16/camss.h
+++ b/drivers/media/platform/qcom/camss-8x16/camss.h
@@ -41,11 +41,11 @@
 	(to_camss(ptr_module)->dev)
 
 #define module_pointer(ptr_module, index)	\
-	((const struct ptr_module##_device (*)[]) &(ptr_module[-(index)]))
+	((const struct ptr_module##_device *) &(ptr_module[-(index)]))
 
 #define to_camss_index(ptr_module, index)	\
 	container_of(module_pointer(ptr_module, index),	\
-		     struct camss, ptr_module)
+		     struct camss, ptr_module[0])
 
 #define to_device_index(ptr_module, index)	\
 	(to_camss_index(ptr_module, index)->dev)
-- 
